src/ServerTlsContext.php

```diff
@@ -166,6 +168,40 @@ public function hasPeerVerification(): bool
         return $this->verifyPeer;
     }
 
+    /**
+     * Enable peer name verification, this is the default with verifyPeer enabled.
+     *
+     * @return self Cloned, modified instance.
+     */
+    public function withPeerNameVerification(): self
+    {
+        $clone = clone $this;
+        $clone->verifyPeerName = true;
+
+        return $clone;
+    }
+
+    /**
+     * Disable peer name verification.
+     *
+     * @return self Cloned, modified instance.
+     */
+    public function withoutPeerNameVerification(): self
+    {
+        $clone = clone $this;
+        $clone->verifyPeerName = false;
+
+        return $clone;
+    }
+
+    /**
+     * @return bool Whether peer verification is enabled.
+     */
+    public function hasPeerNameVerification(): bool
+    {
```

Comment:
Should this really check both properties?

Reply:
IMHO it should, as you should not trust a name from an unverified peer. With this PR, disabling peer verification also disables name verification. My main motivation here was keeping compatibility with the former behaviour. There might of course be use cases asking for name verification w/o verify_peer. Security-wise they have no meaning, as every self-signed cert would be accepted.